### DISCUSSION TITLE
Feature request: ability to change the filename before downloading.

### DISCUSSION THREAD

**Discussion by hakaserver** [BlafKing/sd-civitai-browser-plus#117] [Ideas]
Some models uploaded to Civitai have non-descriptive or plain bad filenames, frequently because some online model training services give the final model a non-descriptive filename and the file is uploaded as is.

Currently, when visualizing a model, the 'Model filename:' field shows the name of the selected file, but there is no option to alter it. I think it would be a great addition and very helpful to be able to manually type a new name in this field to automatically download all files (model info, images, and the model file) with the custom name.

**Comment by BlafKing** [comment]
Sounds good!

Although please do keep in mind that altering the file name has a small chance of breaking some features, since detecting files by filename is a fallback if there is ever an error with detecting the model's SHA256.
So if the code cannot find the model's SHA256 for any reason, then it will attempt to find the model by looking for the matching filename, which will also fail if the filename has been altered.

This should be a rare case though, so it shouldn't be too big of a deal, but worth pointing out regardless :)

**Comment by BlafKing** [comment]
v3.0 now includes the ability to change the filename before saving! :)
